// @react-spectrum/menu/dist/Popover.module.js
import "./menu.57b3a408.css";
import $g5w4B$overlays_cssmodulejs from "./overlays_css.module.js";
import "./menu.2259a533.css";
import $g5w4B$popover_vars_cssmodulejs from "./popover_vars_css.module.js";
import {Underlay as $274a838403463b67$export$f360afc887607b02} from "./Underlay.module.js";
import {usePopover as $87eca3bd2e28ca01$export$542a6fd13ac93354} from "./usePopover.module.js";
import {DismissButton as $g5w4B$DismissButton} from "@react-aria/overlays";
import {useDOMRef as $g5w4B$useDOMRef, useStyleProps as $g5w4B$useStyleProps, classNames as $g5w4B$classNames} from "@react-spectrum/utils";
import {useFocusWithin as $g5w4B$useFocusWithin} from "@react-aria/interactions";
import {useObjectRef as $g5w4B$useObjectRef, mergeProps as $g5w4B$mergeProps, useLayoutEffect as $g5w4B$useLayoutEffect} from "@react-aria/utils";
import {Overlay as $g5w4B$Overlay} from "@react-spectrum/overlays";
import $g5w4B$react, {forwardRef as $g5w4B$forwardRef, useRef as $g5w4B$useRef, useState as $g5w4B$useState} from "react";


function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 









/**
 * Arrow placement can be done pointing right or down because those paths start at 0, x or y. Because the
 * other two don't, they start at a fractional pixel value, it introduces rounding differences between browsers and
 * between display types (retina with subpixels vs not retina). By flipping them with CSS we can ensure that
 * the path always starts at 0 so that it perfectly overlaps the popover's border.
 * See bottom of file for more explanation.
 */ let $bf810f3fb07f8592$var$arrowPlacement = {
    left: 'right',
    right: 'right',
    top: 'bottom',
    bottom: 'bottom'
};
const $bf810f3fb07f8592$export$5b6b19405a83ff9d = /*#__PURE__*/ (0, $g5w4B$forwardRef)(function Popover(props, ref) {
    let { children: children, state: state, ...otherProps } = props;
    let domRef = (0, $g5w4B$useDOMRef)(ref);
    let wrapperRef = (0, $g5w4B$useRef)(null);
    return /*#__PURE__*/ (0, $g5w4B$react).createElement((0, $g5w4B$Overlay), {
        ...otherProps,
        isOpen: state.isOpen,
        nodeRef: wrapperRef
    }, /*#__PURE__*/ (0, $g5w4B$react).createElement($bf810f3fb07f8592$var$PopoverWrapper, {
        ref: domRef,
        ...props,
        wrapperRef: wrapperRef
    }, children));
});
const $bf810f3fb07f8592$var$PopoverWrapper = /*#__PURE__*/ (0, $g5w4B$forwardRef)((props, ref)=>{
    let { children: children, isOpen: isOpen, hideArrow: hideArrow, isNonModal: isNonModal, enableBothDismissButtons: enableBothDismissButtons, state: state, wrapperRef: wrapperRef, onDismissButtonPress: onDismissButtonPress = ()=>state.close() } = props;
    let { styleProps: styleProps } = (0, $g5w4B$useStyleProps)(props);
    let objRef = (0, $g5w4B$useObjectRef)(ref);
    let { size: size, borderWidth: borderWidth, arrowRef: arrowRef } = $bf810f3fb07f8592$var$useArrowSize();
    const borderRadius = $bf810f3fb07f8592$var$usePopoverBorderRadius(objRef);
    let borderDiagonal = borderWidth * Math.SQRT2;
    let primary = size + borderDiagonal;
    let secondary = primary * 2;
    let { popoverProps: popoverProps, arrowProps: arrowProps, underlayProps: underlayProps, placement: placement } = (0, $87eca3bd2e28ca01$export$542a6fd13ac93354)({
        ...props,
        popoverRef: objRef,
        maxHeight: undefined,
        arrowSize: hideArrow ? 0 : secondary,
        arrowBoundaryOffset: borderRadius
    }, state);
    let { focusWithinProps: focusWithinProps } = (0, $g5w4B$useFocusWithin)(props);
    // Attach Transition's nodeRef to outermost wrapper for node.reflow: https://github.com/reactjs/react-transition-group/blob/c89f807067b32eea6f68fd6c622190d88ced82e2/src/Transition.js#L231
    return /*#__PURE__*/ (0, $g5w4B$react).createElement("div", {
        ref: wrapperRef
    }, !isNonModal && /*#__PURE__*/ (0, $g5w4B$react).createElement((0, $274a838403463b67$export$f360afc887607b02), {
        isTransparent: true,
        ...(0, $g5w4B$mergeProps)(underlayProps),
        isOpen: isOpen
    }), /*#__PURE__*/ (0, $g5w4B$react).createElement("div", {
        ...styleProps,
        ...(0, $g5w4B$mergeProps)(popoverProps, focusWithinProps),
        style: {
            ...styleProps.style,
            ...popoverProps.style
        },
        ref: objRef,
        className: (0, $g5w4B$classNames)((0, ($parcel$interopDefault($g5w4B$popover_vars_cssmodulejs))), 'spectrum-Popover', `spectrum-Popover--${placement}`, {
            'spectrum-Popover--withTip': !hideArrow,
            'is-open': isOpen,
            [`is-open--${placement}`]: isOpen
        }, (0, $g5w4B$classNames)((0, ($parcel$interopDefault($g5w4B$overlays_cssmodulejs))), 'spectrum-Popover', 'react-spectrum-Popover'), styleProps.className),
        role: "presentation",
        "data-testid": "popover"
    }, (!isNonModal || enableBothDismissButtons) && /*#__PURE__*/ (0, $g5w4B$react).createElement((0, $g5w4B$DismissButton), {
        onDismiss: onDismissButtonPress
    }), children, hideArrow ? null : /*#__PURE__*/ (0, $g5w4B$react).createElement($bf810f3fb07f8592$var$Arrow, {
        arrowProps: arrowProps,
        isLandscape: placement != null ? $bf810f3fb07f8592$var$arrowPlacement[placement] === 'bottom' : false,
        arrowRef: arrowRef,
        primary: primary,
        secondary: secondary,
        borderDiagonal: borderDiagonal
    }), /*#__PURE__*/ (0, $g5w4B$react).createElement((0, $g5w4B$DismissButton), {
        onDismiss: onDismissButtonPress
    })));
});
function $bf810f3fb07f8592$var$usePopoverBorderRadius(popoverRef) {
    let [borderRadius, setBorderRadius] = (0, $g5w4B$useState)(0);
    (0, $g5w4B$useLayoutEffect)(()=>{
        if (popoverRef.current) {
            let spectrumBorderRadius = window.getComputedStyle(popoverRef.current).borderRadius;
            if (spectrumBorderRadius !== '') setBorderRadius(parseInt(spectrumBorderRadius, 10));
        }
    }, [
        popoverRef
    ]);
    return borderRadius;
}
function $bf810f3fb07f8592$var$useArrowSize() {
    let [size, setSize] = (0, $g5w4B$useState)(20);
    let [borderWidth, setBorderWidth] = (0, $g5w4B$useState)(1);
    let arrowRef = (0, $g5w4B$useRef)(null);
    // get the css value for the tip size and divide it by 2 for this arrow implementation
    (0, $g5w4B$useLayoutEffect)(()=>{
        if (arrowRef.current) {
            let spectrumTipWidth = window.getComputedStyle(arrowRef.current).getPropertyValue('--spectrum-popover-tip-size');
            if (spectrumTipWidth !== '') setSize(parseInt(spectrumTipWidth, 10) / 2);
            let spectrumBorderWidth = window.getComputedStyle(arrowRef.current).getPropertyValue('--spectrum-popover-tip-borderWidth');
            if (spectrumBorderWidth !== '') setBorderWidth(parseInt(spectrumBorderWidth, 10));
        }
    }, []);
    return {
        size: size,
        borderWidth: borderWidth,
        arrowRef: arrowRef
    };
}
function $bf810f3fb07f8592$var$Arrow(props) {
    let { primary: primary, secondary: secondary, isLandscape: isLandscape, arrowProps: arrowProps, borderDiagonal: borderDiagonal, arrowRef: arrowRef } = props;
    let halfBorderDiagonal = borderDiagonal / 2;
    let primaryStart = 0;
    let primaryEnd = primary - halfBorderDiagonal;
    let secondaryStart = halfBorderDiagonal;
    let secondaryMiddle = secondary / 2;
    let secondaryEnd = secondary - halfBorderDiagonal;
    let pathData = isLandscape ? [
        'M',
        secondaryStart,
        primaryStart,
        'L',
        secondaryMiddle,
        primaryEnd,
        'L',
        secondaryEnd,
        primaryStart
    ] : [
        'M',
        primaryStart,
        secondaryStart,
        'L',
        primaryEnd,
        secondaryMiddle,
        'L',
        primaryStart,
        secondaryEnd
    ];
    /* use ceil because the svg needs to always accommodate the path inside it */ return /*#__PURE__*/ (0, $g5w4B$react).createElement("svg", {
        xmlns: "http://www.w3.org/svg/2000",
        width: Math.ceil(isLandscape ? secondary : primary),
        height: Math.ceil(isLandscape ? primary : secondary),
        className: (0, $g5w4B$classNames)((0, ($parcel$interopDefault($g5w4B$popover_vars_cssmodulejs))), 'spectrum-Popover-tip'),
        ref: arrowRef,
        ...arrowProps
    }, /*#__PURE__*/ (0, $g5w4B$react).createElement("path", {
        className: (0, $g5w4B$classNames)((0, ($parcel$interopDefault($g5w4B$popover_vars_cssmodulejs))), 'spectrum-Popover-tip-triangle'),
        d: pathData.join(' ')
    }));
}


export {$bf810f3fb07f8592$export$5b6b19405a83ff9d as Popover};
